backend/dataall/core/environment/services/environment_service.py

```diff
@@ -295,6 +296,12 @@ def create_environment(uri, data=None):
             )
             session.commit()
 
+            ResourceLockRepository.create_resource_lock(
+                session=session,
+                resource_uri=f'{env_group.groupUri}-{env_group.environmentUri}',
```

Comment:
don't we assume that groupUris are unique? why do you need to append the envUri?

Reply:
The same group can be a memeber of many different environments

In practice primary key for `environment_group_permission` (aka EnvironmentGroup) is groupUri + environmentUri

We similarly need a way to distinguish each Env + Group Pair for resource lock since the underlying resource we are "locking" is the consumer IAM role for the group in that particular env